android/src/main/kotlin/com/github/s0nerik/fast_contacts/FastContactsPlugin.kt

```diff
@@ -137,40 +134,97 @@ class FastContactsPlugin : FlutterPlugin, MethodCallHandler, LifecycleOwner, Vie
             val contactId = cursor.getLong(projection.indexOf(Phone.CONTACT_ID))
             val displayName = cursor.getString(projection.indexOf(Phone.DISPLAY_NAME)) ?: ""
             val phone = cursor.getString(projection.indexOf(Phone.NUMBER)) ?: ""
+            val type = cursor.getInt(projection.indexOf(Phone.TYPE))
+            val customLabel = cursor.getString(projection.indexOf(Phone.LABEL))
+
+            val label = customLabel ?: getPhoneLabel(type)
 
             if (contacts.containsKey(contactId)) {
-                (contacts[contactId]!!.phones as MutableList<String>).add(phone)
+                (contacts[contactId]!!.phones as MutableList<ContactPhone>).add(ContactPhone(
+                    number = phone,
+                    label = label
+                ))
             } else {
                 contacts[contactId] = Contact(
                         id = contactId.toString(),
                         displayName = displayName,
-                        phones = mutableListOf(phone)
+                        phones = mutableListOf(ContactPhone(
+                            number = phone,
+                            label = label
+                        ))
                 )
             }
         }
         return contacts
     }
 
+    private fun getPhoneLabel(type: Int): String {
+        return when (type) {
+            Phone.TYPE_ASSISTANT -> "assistant"
+            Phone.TYPE_CALLBACK -> "callback"
+            Phone.TYPE_CAR -> "car"
+            Phone.TYPE_COMPANY_MAIN -> "companyMain"
+            Phone.TYPE_FAX_HOME -> "faxHome"
+            Phone.TYPE_FAX_WORK -> "faxWork"
+            Phone.TYPE_HOME -> "home"
+            Phone.TYPE_ISDN -> "isdn"
+            Phone.TYPE_MAIN -> "main"
+            Phone.TYPE_MMS -> "mms"
+            Phone.TYPE_MOBILE -> "mobile"
+            Phone.TYPE_OTHER -> "other"
+            Phone.TYPE_OTHER_FAX -> "faxOther"
+            Phone.TYPE_PAGER -> "pager"
+            Phone.TYPE_RADIO -> "radio"
+            Phone.TYPE_TELEX -> "telex"
+            Phone.TYPE_TTY_TDD -> "ttyTtd"
+            Phone.TYPE_WORK -> "work"
+            Phone.TYPE_WORK_MOBILE -> "workMobile"
+            Phone.TYPE_WORK_PAGER -> "workPager"
+            Phone.TYPE_CUSTOM -> "custom"
+            else -> "mobile"
```

Comment:
IIUC, this might potentially become invalid as at this point the phone type is actually unknown, but we're treating it as "mobile".

I imagine this could easily become an issue if OS adds another phone type in a newer version, but the plugin is not yet updated to support it.

It would be much better, IMO, to have `""` returned here and let the user decide how to treat it. Looks like iOS implementation follows this path already.

Reply:
Yeah, u're right. Ill patch it up in a moment